### PR TITLE
Configure dev-tools to be reusable by community beats

### PR DESCRIPTION
**Pull request by cyrilleverrier**
1. Split `dev-tools/packer/Makefile` into `dev-tools/packer/Makefile` and `dev-tools/packer/scripts/Makefile` (reusable Makefile)

2. Fix paths in scripts and Makefiles

3. Distinguish between official beats (inside `github.com/elastic/beats/topbeat`) and community beats (inside `github.com/my-github-name/examplebeat`) when fixing paths

Note:
1. See https://github.com/cyrilleverrier/beat-generator/commit/81e6436a580d9199ca31f15f9dd6f492476f1697 for an example on how to integrate it in a community beat
2. This is the least worst solution I could implement in order to reuse the dev-tools scripts.
The script `xgo-image/base/build.sh` is no longer generic ("Beats" specific) and quite different from its original source at https://github.com/karalabe/xgo/tree/master/docker. But it was already heavily modified so I believe this is an acceptable solution.

### CODE REVIEW

File: dev-tools/packer/scripts/Makefile [hunk 1]
```diff
@@ -0,0 +1,124 @@
+DATE:=$(shell date +%y%m%d%H%M%S)
+BUILDID?=$(DATE)
+
+makefile_abspath:=$(abspath $(lastword $(MAKEFILE_LIST))/..)
+packer_absdir=$(shell dirname ${makefile_abspath})
+
+
+%/deb: % build/god-linux-386 build/god-linux-amd64 fpm-image
```
Comment (ruflin) [inline dev-tools/packer/scripts/Makefile]: I assume that is a 1-1 copy of the previously used Makefile (from line 28)

Comment (cyrilleverrier) [inline dev-tools/packer/scripts/Makefile]: I split the original Makefile into 2 Makefiles and added $(packer_absdir) to fix path locations

**Comment by ruflin** [comment]
LGTM. @tsg @andrewkroh Could one of you also have a brief look at it as you worked more with the dev-tools in the past.

**Comment by andrewkroh** [comment]
LGTM, I think the best way to validate this type of change is to run it. @cyrilleverrier have you tested that when the [`deploy`](https://github.com/elastic/beats/blob/master/dev-tools/deploy) runs that all the binaries are produced in `dev-tools/packer/build/binary/upload/<beat>/` same as before?

As soon as this merges it will be tested by the Jenkins [beats-package](http://build-eu-00.elastic.co/view/Beats/job/beats-package/) job, but I would be nice to know before it merges that it still works. :smile:

**Comment by cyrilleverrier** [comment]
Test performed:

1. Use  https://github.com/cyrilleverrier/beat-generator/commit/81e6436a580d9199ca31f15f9dd6f492476f1697
 to generate code  `github.com/your-github-name/examplebeat`

2. Checkout `github.com/your-github-name/examplebeat/vendor/github.com/elastic/beats` to get this PR

3. Create xgo docker images and all standard beat packages:
```
cd /home/cyrille/code/go/workspace/src/github.com/your-github-name/examplebeat/vendor/github.com/elastic/beats/dev-tools

./deploy
```

4. Create `examplebeat` packages:
```
cd /home/cyrille/code/go/workspace/src/github.com/your-github-name/examplebeat/dev-tools/packer
make
```

**Comment by andrewkroh** [comment]
Thanks @cyrilleverrier! I think we should merge this right after we cut the 5.0.0-alpha1 release just to be safe.

**Comment by ruflin** [comment]
@andrewkroh Alternatively we could see 5.0.0-alpha1 as a good test bed for this :-)

**Comment by andrewkroh** [comment]
I'm OK with merging now. Is @tsg?

**Comment by tsg** [comment]
@andrewkroh @cyrilleverrier Generally yes, but I'd like to test it on my computer first just to make sure we can still release 5.0-alpha1 :-). Will report back.

**Comment by tsg** [comment]
All good, thank you @cyrilleverrier!

**Comment by cyrilleverrier** [comment]
thanks for merging @tsg  and @ruflin

**Pull request merged**